### PR TITLE
#162633460 Implement User login

### PR DESCRIPTION
**Pull request by fatukunda**
**What does this PR do?**

Implements login functionality

**Description of Task to be completed?**
- Create a Login component.
- Create action, reducer, and redux store files.
- Connect the redux store with the login component.

**How should this be manually tested?**

 - Clone the project by running `git clone https://github.com/andela/ah-frontend-valkyrie.git`
 - After cloning the repo, `cd` into it and run `npm install` to install dependencies.
- Run `npm start` to run the application.
- Run `npm test` to run the tests for the application.
- Click on login button in the navigation bar.
- Fill in the form with a valid username and password.

**What are the relevant pivotal tracker stories?**

[#162633460](https://www.pivotaltracker.com/story/show/162633460)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,25 +15,29 @@
...
   "license": "ISC",
   "dependencies": {
+    "@types/moxios": "^0.4.8",
+    "axios": "^0.18.0",
     "babel-jest": "^23.4.2",
     "babel-loader": "^8.0.5",
     "css-loader": "^2.1.0",
     "enzyme": "^3.3.0",
     "enzyme-adapter-react-16": "^1.1.1",
-    "eslint-config-airbnb": "^17.1.0",
```
Comment (mubstimor) [inline package.json:31]: Why did you remove this linter?

File: src/App.js [hunk 2]
```diff
@@ -1,12 +1,21 @@
...
 import Landing from './components/layout/Landing';
 import Login from './components/auth/Login';
 import Register from './components/auth/Register';
+import { loginSuccess, setCurrentUser } from './actions/loginActions';
 
+if ( localStorage.auth_token ) {
+  const userData = jwt_decode( localStorage.auth_token );
+  store.dispatch(setCurrentUser( true ));
+  store.dispatch(loginSuccess( userData ));
+
```
Comment (mubstimor) [inline src/App.js]: Blank line not needed here

**Comment by mubstimor** [comment]
Please squash the commits to one and resolve the small issues I flagged, otherwise LGTM.

**Comment by Qadriyah** [comment]
Resolve hound issues

File: package.json [hunk 3]
```diff
@@ -15,25 +15,29 @@
     "test:w": "jest --watch --coverage",
     "test:coverage": "jest --coverage --colors",
     "eject": "react-scripts eject",
-    "start": "node server.js",
+    "start": "webpack-dev-server --mode development",
```
Comment (jeanjoe) [inline package.json]: Please add the node server for heroku to run

**Review by jeanjoe — CHANGES_REQUESTED** [review]

File: src/components/auth/Login.js [hunk 4]
```diff
@@ -1,46 +1,103 @@
...
-                value="Sign in"
+                type='submit'
+                className='btn btn-block mt-4 btn-login'
+                value='Sign in'
               />
+              {isLoginPending && <div className="spinner-border text-info" role="status">
+                <span className="sr-only">Loading...</span>
+              </div>}
+              {loginSuccess && <Link to="/users/profile" />}
+              {isAuthenticated && console.log('Authenticated')}
```
Comment (NabunyaLilian) [inline src/components/auth/Login.js]: Please remove the console log

File: src/components/auth/styles/Login.css [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    margin-top: 20px;
+ }
+ .twitter-btn{
+     background-color: #00a2f8;
+     color: #fff;
+ }
+ .facebook-btn{
+     background-color:  #3b5998;
+     color: #fff;
+ }
```
Comment (NabunyaLilian) [inline src/components/auth/styles/Login.css:35]: Please add a new line

File: src/components/layout/styles/Navbar.css [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    height: 32px;
+    width: 32px;
+    border-radius: 50%;
+    cursor: pointer;
+}
+.notification {
+    height: 32px;
+    width: 32px;
+    color: #fff;
+}
```
Comment (NabunyaLilian) [inline src/components/layout/styles/Navbar.css:19]: New line to the end of the file is needed.

File: src/index.html [hunk 7]
```diff
@@ -24,20 +24,27 @@
...
+    crossorigin="anonymous"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.7/umd/popper.min.js"
+    integrity="sha384-UO2eT0CpHqdSJQ6hJty5KVphtPhzWj9WO1clHTMGa3JDZwrnQq4sF86dIHNDz0W1"
+    crossorigin="anonymous"></script>
+  <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.0/js/bootstrap.min.js"
+    integrity="sha384-7aThvCh9TypR7fIc2HV4O/nFMVCBwyIUKL8XCtKE+8xgCgl/PQGuFsvShjr74PBp"
+    crossorigin="anonymous"></script>
+</body>
+
+</html>
```
Comment (NabunyaLilian) [inline src/index.html]: Same here, new line

**Comment by NabunyaLilian** [comment]
Good work Frank though you should resolve the violations.

File: .babelrc [hunk 8]
```diff
@@ -1,3 +1,9 @@
...
-  "presets": ["@babel/preset-env", "@babel/preset-react"]
-}
+  "presets": [
+    "@babel/preset-env",
+    "@babel/preset-react"
+  ],
+  "plugins": [
+    "@babel/plugin-proposal-class-properties"
+  ]
+}
```
Comment (ruganda) [inline .babelrc:9]: Please add a new line at the end of the file

File: package.json [hunk 9]
```diff
@@ -43,28 +47,46 @@
...
+    "sass-loader": "^7.1.0",
+    "sfcookies": "^1.0.2",
+    "style-loader": "^0.23.1",
     "typescript": "^3.3.3",
     "webpack": "^4.29.3",
     "webpack-cli": "^3.2.3",
-    "webpack-dev-server": "^3.1.14"
+    "webpack-dev-server": "^3.1.14",
+    "sass-loader": "^7.1.0",
+    "style-loader": "^0.23.1"
```
Comment (jeanjoe) [inline package.json]: Please remove this duplicate module, it already exists on line 56
